### PR TITLE
Add retry logic for RTC signaling failure.

### PR DESCRIPTION
**Pull request by supelee**
*Description of changes:*
This change adds a configurable number of retries for states in the RtcSession state machine context, and configures 3 retries for the ConnectSignalingAndIceCollectionState.

### CODE REVIEW

File: uml/Session.puml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+[*] --> GrabLocalMedia
+GrabLocalMedia --> CreateOffer
+GrabLocalMedia --> Failed
+CreateOffer --> SetLocalSessionDescription
+CreateOffer --> Failed
+SetLocalSessionDescription --> CollectSignalingAndIceCollection
+SetLocalSessionDescription --> Failed
+CollectSignalingAndIceCollection --> InviteAnswer
+CollectSignalingAndIceCollection --> CollectSignalingAndIceCollection : RETRY
```
Comment (supelee) [inline uml/Session.puml]: This is no longer true, will remove.

File: test/unit/rtc_session.js [hunk 2]
```diff
@@ -291,11 +291,24 @@ describe('RTC session', () => {
         var state;
 
         beforeEach(() => {
+            var retryContext = {
```
Comment (yuliangz) [inline test/unit/rtc_session.js]: Revert this file?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+sudo: false
```
Comment (yuliangz) [inline .travis.yml]: Let's remove this file, we will get to travis later

File: uml/Signaling.puml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+@startuml
+
+[*] --> PendingConnection
+PendingConnection --> PendingInvite
```
Comment (yuliangz) [inline uml/Signaling.puml]: Now there is a loop from PendingConnection to PendingConnection

**Comment by supelee** [comment]
Merged into master branch, closing pull request.

**Pull request closed without merging**